docs: New documentation (#858)
dbbs-lab/bsb-core: docs/topology/partitions.rst

Partitions
##########

Partitions contain shape descriptions used for the spatial layout of the network.
These descriptions can be represented as layers, meshes and voxelsets.

> In BSB are implemented two different kind of partitions: :ref:`layer  <layer-partition>` and :ref:`voxel based <voxel-partition>`.

Reviewer: ```suggestion
In the BSB the following two partitions are implemented: 
* :ref:`Layers  <layer-partition>` 
* :ref:`Voxelized volumes <voxel-partition>`.
```

.. _layer-partition:

======
Layers
======

:class:`Layer partition <.topology.partition.Layer>` is a parallelepiped shaped volume.
The size of a layer is defined by the network except for the height, therefore it is needed to

> specify the thickness of the layer with the ``thickness`` attribute.

Reviewer: GPT

.. tab-set-code::

    .. code-block:: json

      {
        "partitions": {
          "my_layer_partition": {
            "type": "layer",
            "thickness": 25
          }
        }
      }

Cortical Model Use Case
-----------------------

Layers can be organized in a pile in order to separate the structure of your network according to the depth.

> :ref:`Stack <stack-region>` region is used to pile up your layers, for a three layer system it will be:

Reviewer: GPT
